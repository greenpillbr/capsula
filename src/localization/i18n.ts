import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import { APP_CONFIG } from '../constants'

// Import localization files
import enUS from './locales/en-US'
import esES from './locales/es-ES'
import ptBR from './locales/pt-BR'

const resources = {
  'pt-BR': {
    translation: ptBR
  },
  'en-US': {
    translation: enUS
  },
  'es-ES': {
    translation: esES
  },
}

i18n
  .use(initReactI18next)
  .init({
    resources,
    lng: APP_CONFIG.defaultLanguage,
    fallbackLng: 'en-US',
    
    interpolation: {
      escapeValue: false,
      format: (value, format, lng) => {
        if (format === 'currency') {
          return formatCurrency(value, lng || APP_CONFIG.defaultLanguage)
        }
        if (format === 'date') {
          return formatDate(value, lng || APP_CONFIG.defaultLanguage)
        }
        if (format === 'number') {
          return formatNumber(value, lng || APP_CONFIG.defaultLanguage)
        }
        return value
      }
    },
    
    react: {
      useSuspense: false,
    },
  })

// Brazilian-specific formatting functions
export const formatCurrency = (amount: number, locale: string = 'pt-BR'): string => {
  if (locale === 'pt-BR') {
    // Format as Brazilian Real
    return new Intl.NumberFormat('pt-BR', {
      style: 'currency',
      currency: 'BRL'
    }).format(amount)
  }
  
  if (locale === 'en-US') {
    return new Intl.NumberFormat('en-US', {
      style: 'currency',
      currency: 'USD'
    }).format(amount)
  }
  
  if (locale === 'es-ES') {
    return new Intl.NumberFormat('es-ES', {
      style: 'currency',
      currency: 'EUR'
    }).format(amount)
  }
  
  return amount.toString()
}

export const formatCrypto = (amount: number, symbol: string): string => {
  const formatted = amount.toFixed(6).replace(/\.?0+$/, '')
  return `${formatted} ${symbol}`
}

export const formatDate = (date: Date, locale: string = 'pt-BR'): string => {
  const options: Intl.DateTimeFormatOptions = {
    day: '2-digit',
    month: '2-digit',
    year: 'numeric',
    hour: '2-digit',
    minute: '2-digit'
  }
  
  return new Intl.DateTimeFormat(locale, options).format(date)
}

export const formatNumber = (number: number, locale: string = 'pt-BR'): string => {
  return new Intl.NumberFormat(locale).format(number)
}

export const formatAddress = (address: string, length: number = 6): string => {
  if (!address) return ''
  if (address.length <= length * 2) return address
  
  return `${address.slice(0, length)}...${address.slice(-length)}`
}

export default i18n
