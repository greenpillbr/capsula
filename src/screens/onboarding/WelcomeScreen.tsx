import { Colors, LAYOUT } from '@/constants'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'

interface WelcomeScreenProps {
  onGetStarted: () => void
}

export const WelcomeScreen: React.FC<WelcomeScreenProps> = ({ onGetStarted }) => {
  const { t } = useTranslation()

  return (
    <View style={styles.container}>
      <View style={styles.content}>
        {/* Logo placeholder */}
        <View style={styles.logoContainer}>
          <Text style={styles.logoText}>ðŸŒ±</Text>
          <Text style={styles.appName}>Capsula</Text>
        </View>

        {/* Welcome content */}
        <View style={styles.textContainer}>
          <Text style={styles.title}>{t('onboarding.welcome.title')}</Text>
          <Text style={styles.subtitle}>{t('onboarding.welcome.subtitle')}</Text>
          <Text style={styles.description}>{t('onboarding.welcome.description')}</Text>
        </View>

        {/* Action button */}
        <TouchableOpacity style={styles.button} onPress={onGetStarted}>
          <Text style={styles.buttonText}>{t('common.continue')}</Text>
        </TouchableOpacity>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.background,
  },
  content: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: LAYOUT.padding.lg,
  },
  logoContainer: {
    alignItems: 'center',
    marginBottom: LAYOUT.padding.xl * 2,
  },
  logoText: {
    fontSize: 80,
    marginBottom: LAYOUT.padding.md,
  },
  appName: {
    fontSize: LAYOUT.fontSize.xxl,
    fontWeight: 'bold',
    color: Colors.primary,
  },
  textContainer: {
    alignItems: 'center',
    marginBottom: LAYOUT.padding.xl * 2,
  },
  title: {
    fontSize: LAYOUT.fontSize.xxl,
    fontWeight: 'bold',
    color: Colors.text,
    textAlign: 'center',
    marginBottom: LAYOUT.padding.md,
  },
  subtitle: {
    fontSize: LAYOUT.fontSize.lg,
    color: Colors.textSecondary,
    textAlign: 'center',
    marginBottom: LAYOUT.padding.lg,
  },
  description: {
    fontSize: LAYOUT.fontSize.md,
    color: Colors.textSecondary,
    textAlign: 'center',
    lineHeight: 24,
  },
  button: {
    backgroundColor: Colors.primary,
    paddingHorizontal: LAYOUT.padding.xl,
    paddingVertical: LAYOUT.padding.lg,
    borderRadius: LAYOUT.borderRadius.lg,
    minWidth: 200,
  },
  buttonText: {
    color: Colors.surface,
    fontSize: LAYOUT.fontSize.lg,
    fontWeight: '600',
    textAlign: 'center',
  },
})
