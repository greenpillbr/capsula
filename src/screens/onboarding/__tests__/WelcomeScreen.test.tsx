import '@/localization/i18n'; // Initialize i18n
import { fireEvent, render } from '@testing-library/react-native'
import { WelcomeScreen } from '../WelcomeScreen'

// Mock the translation hook
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: string) => {
      const translations: Record<string, string> = {
        'onboarding.welcome.title': 'Welcome to Capsula',
        'onboarding.welcome.subtitle': 'Your regenerative wallet for the future economy',
        'onboarding.welcome.description': 'Manage your digital assets and participate in regenerative communities simply and securely.',
        'common.continue': 'Continue',
      }
      return translations[key] || key
    },
  }),
}))

describe('WelcomeScreen', () => {
  const mockOnGetStarted = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('renders correctly', () => {
    const { getByText } = render(<WelcomeScreen onGetStarted={mockOnGetStarted} />)
    
    expect(getByText('Capsula')).toBeTruthy()
    expect(getByText('Welcome to Capsula')).toBeTruthy()
    expect(getByText('Your regenerative wallet for the future economy')).toBeTruthy()
    expect(getByText('Manage your digital assets and participate in regenerative communities simply and securely.')).toBeTruthy()
    expect(getByText('Continue')).toBeTruthy()
  })

  it('displays the logo emoji', () => {
    const { getByText } = render(<WelcomeScreen onGetStarted={mockOnGetStarted} />)
    
    expect(getByText('ðŸŒ±')).toBeTruthy()
  })

  it('calls onGetStarted when continue button is pressed', () => {
    const { getByText } = render(<WelcomeScreen onGetStarted={mockOnGetStarted} />)
    
    const continueButton = getByText('Continue')
    fireEvent.press(continueButton)
    
    expect(mockOnGetStarted).toHaveBeenCalledTimes(1)
  })

  it('has proper accessibility', () => {
    const { getByText } = render(<WelcomeScreen onGetStarted={mockOnGetStarted} />)
    
    const continueButton = getByText('Continue')
    expect(continueButton).toBeTruthy()
    
    // The button should be touchable
    fireEvent.press(continueButton)
    expect(mockOnGetStarted).toHaveBeenCalled()
  })

  it('matches snapshot', () => {
    const tree = render(<WelcomeScreen onGetStarted={mockOnGetStarted} />)
    expect(tree).toMatchSnapshot()
  })
})
