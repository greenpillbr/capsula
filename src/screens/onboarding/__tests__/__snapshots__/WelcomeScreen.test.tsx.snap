// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#FAF9F6",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "marginBottom": 64,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 80,
            "marginBottom": 16,
          }
        }
      >
        ðŸŒ±
      </Text>
      <Text
        style={
          {
            "color": "#2D5A27",
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Capsula
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "marginBottom": 64,
        }
      }
    >
      <Text
        style={
          {
            "color": "#2C3E50",
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 16,
            "textAlign": "center",
          }
        }
      >
        Welcome to Capsula
      </Text>
      <Text
        style={
          {
            "color": "#7F8C8D",
            "fontSize": 18,
            "marginBottom": 24,
            "textAlign": "center",
          }
        }
      >
        Your regenerative wallet for the future economy
      </Text>
      <Text
        style={
          {
            "color": "#7F8C8D",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "center",
          }
        }
      >
        Manage your digital assets and participate in regenerative communities simply and securely.
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#2D5A27",
          "borderRadius": 12,
          "minWidth": 200,
          "opacity": 1,
          "paddingHorizontal": 32,
          "paddingVertical": 24,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        Continue
      </Text>
    </View>
  </View>
</View>
`;
